### PR TITLE
Support CI for Windows and MacOS

### PR DESCRIPTION
**Pull request by peternied**
### Description
Creates a github action to install and configure Dashboards that is reused within the workflows of this repository, its multi-platform safe.

### Issues Resolved
- Related https://github.com/opensearch-project/security-dashboards-plugin/issues/845
- Related https://github.com/opensearch-project/security-dashboards-plugin/issues/1156

Note: follow up pull request will convert the integration test job to be multi-platform

### Check List
- [X] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by davidlago — APPROVED** [review]
Nice refactoring, thanks for taking care of this

**Comment by cwperks** [comment]
@peternied Looks like we have to modify the required checks similar to what @cliu123 did for this PR: https://github.com/opensearch-project/security/pull/2161#issuecomment-1294205866

File: .github/actions/install-dashboards/action.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    description: "The directory where the plugin has been configured"
+    value: ${{ steps.determine-plugin-directory.outputs.plugin-directory }}
+
+
+runs:
+  using: "composite"
+  steps:
+    - id: determine-dashboards-directory
+      run: echo "dashboards-directory=OpenSearch-Dashboards"  >> $GITHUB_OUTPUT
+      shell: bash
```
Comment (cwperks) [inline .github/actions/install-dashboards/action.yml:24]: Looks like `bash` is available on all platforms according to this: https://dev.to/pwd9000/github-actions-all-the-shells-581h

Nice use of a composite action! 🙌

File: .github/workflows/unit-test.yml [hunk 2]
```diff
@@ -3,78 +3,35 @@ name: Unit Tests
...
-      - name: Setup node
-        uses: actions/setup-node@v1
+      - name: Enable longer filenames
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: git config --system core.longpaths true
+
+      - uses: actions/checkout@v2
+
+      - id: install-dashboards
+        uses: ./.github/actions/install-dashboards
```
Comment (cwperks) [inline .github/workflows/unit-test.yml:22]: +1 for re-usability!

**Review by cwperks — APPROVED** [review]
Great work @peternied!

**Comment by peternied** [comment]
FYI The backport to 2.x https://github.com/opensearch-project/security-dashboards-plugin/pull/1167

**Pull request merged**